### PR TITLE
Bug when cart empty

### LINKED ISSUE #20496: Bug when cart empty
#### Describe the bug

When cart empty i want call method for get total cart

#### Expected behavior

Return value without notice

#### Steps to Reproduce

1. Create new cart
2. Call method getOrderTotal

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.4.4
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by fest73**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | When cart empty i want call method for get total cart
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/20496.
| How to test?  | Create new cart. Call method getOrderTotal

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20497)

### CODE REVIEW

File: classes/Cart.php [hunk 1]
```diff
@@ -3704,7 +3704,7 @@ public function getTotalWeight($products = null)
             $this->updateProductWeight($this->id);
         }
 
-        return self::$_totalWeight[(int) $this->id];
+        return isset(self::$_totalWeight[(int) $this->id]) ? self::$_totalWeight[(int) $this->id] : 0;
```
Comment (Progi1984) [inline classes/Cart.php]:
```suggestion
        return (float) (self::$_totalWeight[(int) $this->id] ?? 0);
```

File: classes/Cart.php [hunk 2]
```diff
@@ -3704,7 +3704,7 @@ public function getTotalWeight($products = null)
             $this->updateProductWeight($this->id);
         }
 
-        return self::$_totalWeight[(int) $this->id];
+        return (float) (self::$_totalWeight[(int) $this->id] ?? 0);
```
Comment (PierreRambaud) [inline classes/Cart.php]:
```suggestion
        return self::$_totalWeight[(int) $this->id] ?? 0;
```

Float conversation has no aim here :thinking:

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by SimonGrn** [comment]
Hello all,

the linked issue was never validated and the OP doesn't answer. We cannot validate this PR.

**Comment by PierreRambaud** [comment]
Hi @dariusakafest  ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**